profiles/seccomp/seccomp.go

```diff
@@ -6,28 +6,64 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/docker/docker/pkg/stringutils"
 	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/versions/v1p24"
 	"github.com/opencontainers/specs/specs-go"
+	libseccomp "github.com/seccomp/libseccomp-golang"
 )
 
 //go:generate go run -tags 'seccomp' generate.go
 
 // GetDefaultProfile returns the default seccomp profile.
 func GetDefaultProfile(rs *specs.Spec) (*specs.Seccomp, error) {
-	return setupSeccomp(DefaultProfile(rs))
+	return setupSeccomp(DefaultProfile(), rs)
 }
 
 // LoadProfile takes a file path and decodes the seccomp profile.
-func LoadProfile(body string) (*specs.Seccomp, error) {
-	var config types.Seccomp
+func LoadProfile(body string, rs *specs.Spec) (*specs.Seccomp, error) {
+	var (
+		config types.Seccomp
+		err    error
+	)
 	if err := json.Unmarshal([]byte(body), &config); err != nil {
 		return nil, fmt.Errorf("Decoding seccomp profile failed: %v", err)
 	}
+	// try with the old profile
+	// TODO(runcom): this is not going to work properly:
+	// if I use Version == 0 in a new format seccomp profile, what's going to
+	// happen?
+	if config.Version == 0 {
```

Comment:
If you run with this, it'll default to NO syscalls because `syscalls` is there but the contained struct isn't defining names, but it unmarshal ok
```
{
	"version": 1,
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"name": "chmod",
			"action": "SCMP_ACT_ERRNO"
		},
		{
			"name":"fchmod",
			"action": "SCMP_ACT_ERRNO"
		},
		{
			"name": "fchmodat",
			"action":"SCMP_ACT_ERRNO"
		}
	]
}
```